controller/src/main/java/io/pravega/controller/metrics/StreamMetrics.java

```diff
@@ -38,13 +39,30 @@
 /**
  * Class to encapsulate the logic to report Controller service metrics for Streams.
  */
-public final class StreamMetrics extends AbstractControllerMetrics implements AutoCloseable {
+public final class StreamMetrics extends AbstractControllerMetrics {
+
+    private static final AtomicReference<StreamMetrics> INSTANCE = new AtomicReference<>();
```

Comment:
minor nit: since instance is protected by lock (synchronized methods), we can make it non final and remove atomic reference which is basically double locking.

Reply:
I observed that our code checks specifically forbid to create non-final static variables for safety. That's the reason to create an `AtomicReference` there.